docs/upgrading/migrating-local-extensions-33.md

```diff
@@ -0,0 +1,138 @@
+decide to move extensions to be part of your main App namespace and if so you
+can do something like the following: 
+
+```json
+    "autoload" : {
+        "psr-4" : {
+            "MyCompany\\MyApp\\": "src"
+        }
+    }
+```
+
+With the above setup you can move local extensions to `src/Extension/` and
+access the class via `new MyCompany\MyApp\Extension\SuperCoolExtension()`
+
+### Activation
+
+Once your extensions can be autoloaded then one more step is needed to direct
+Bolt to load them.
+
+If you have a `.bolt.yml` file at the root of your project you can define which
+extension classes are loaded within that file. If you open the file as it is now
+then it will look something like this:
+
+```yaml
+
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+```
+
+You will need to edit that file to list which extensions are loaded, note that
+it is the main  extension class that needs to be specified. Assuming the
+autoload setup discussed in the last section your new `.bolt.yml` will now look
+like:
+
+```yaml
+
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+
+extensions:
+    - Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass
+```
+
+To clarify, the value you put in the yml file is exactly what you would use to
+instantiate the  class, so in code the above is equivalent to 
+`new Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass()`
+
+If you used the other autoload structure discussed, your file would look like
+this:
+
+```yaml
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+
+extensions:
+    - MyCompany\MyApp\Extension\SuperCoolExtension
+```
+
+If you use a php file to bootstrap your Bolt application then you will also need
+to add each extension using php code. Your bootstrap file is at `.bolt.php` and
+it currently probably looks like this:
+
+```php
+$configuration = new MyApp\Configuration(__DIR__);
+$app = new Bolt\Application(array('resources'=>$configuration));
+// Any other application setup carries on here…
+
+// Adding your extension to Bolt
+$app['extensions']->add(new Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass());
+
+$app->initialize();
+
+return $app;
+```
+
+## Other Information
+
+As you may have realised, the `composer.json` files are now no longer needed for
+local extensions writing a Bolt extension is as simple as creating a class in
+your Application that implements  `Bolt\Extension\ExtensionInterface` and adding
+the Fully Qualified Class Name to your `.bolt.yml` file.
+
+Once you have a local extension loaded in your application, the interface within
+the Bolt extensions screen has also been adjusted to separate local extensions
+from those installed via the Market Place. These local extensions will now
+appear underneath the other installed extensions.
```

Comment:
This provides a lot of reasoning for the change. IMO this should be moved to the top to explain ourselves.